cart/test/api.test.js

```diff
@@ -0,0 +1,213 @@
+/* eslint-env mocha */
+const chai = require("chai");
+const { expect } = chai;
+const host = "localhost";
+const port = 3000;
+const cartURL = `http://${host}:${port}/cart`; // URL for GraphQL API Gateway
+const requestCart = require("supertest")(cartURL);
+const logger = require('../src/utils/logger');
+
+describe("Cart REST API", () => {
+  let product, cartid, cart, sample_products, sample_users;
+
+  // Utility function to initialize data
+  async function loadSampleData() {
+    // Set up test data
+    sample_users = [
+      {
+        "uid": "user1",
+        "fname": "John",
+        "lname": "Doe",
+        "address": "Some Street 22",
+        "phone": 1231231231,
+        "email": "john@doe.com"
+      }
+    ]
+
+    sample_products = [
+      {
+        "pid": 1,
+        "pcode": "ABC123",
+        "price": 42.5,
+        "sku": "XYZ",
+        "amount_in_stock": 10,
+        "pname": "Something",
+        "description": "This is something very interesting that you want to buy.",
+        "lang": "en_US"
+      },
+      {
+        "pid": 2,
+        "pcode": "FD2",
+        "price": 99.99,
+        "sku": "QWOP",
+        "amount_in_stock": 100,
+        "pname": "Speedos",
+        "description": "I'm too lazy to write a description",
+        "lang": "en_US"
+      }
+    ]
+
+    // Create the records in the database
+    logger.debug(`Inserting sample records`);
+    await cart.deleteAll(); // Clear out the existing data
+    await cart.repository.knex('products').insert(sample_products);
+    await cart.repository.knex('users').insert(sample_users);
+  }
+
+  // Runs before all tests
+  before(function before() {
+
+    // Load the config and wait for it to load
+    const config = require('../src/configs/config');
+    config.then(() => {
+
```

Comment:
let's think about creating a separate method/file for this initial setup

Reply:
OK